include/picongpu/simulation/stage/AtomicPhysics.x.cpp

```diff
@@ -584,10 +581,12 @@ namespace picongpu::simulation::stage
                     recordChanges(mappingDesc);
                     updateElectrons(mappingDesc, currentStep);
                     updateElectricField(mappingDesc);
-                    doIPDIonization(mappingDesc, currentStep, deviceLocalReduce);
                     updateTimeRemaining(mappingDesc);
                     isSubSteppingComplete = isSubSteppingFinished(mappingDesc, deviceLocalReduce);
                 } // end atomicPhysics sub-stepping loop
+
+                // ensure no unbound states are visible to the rest of the loop
+                doIPDIonization(mappingDesc, currentStep, deviceLocalReduce);
```

Comment:
Does the mean we execute the IPD ionization twice per pic time step? Because we call this already in line 540.

Reply:
yes, that is intended and necessary.

The second call of `doIPDIonization` is necessary here since we need to make sure that we do not pass an unbound state with a too low charge state to the pusher and we need to call `doIPDIonization` again at the beginning of the next AtomicPhyiscs step since the we updated the field in between causing a possible change result of the call.

Also note calling `doIPDIonization` twice on the same particle set with an unchanged E-Field does not change particle set anymore after the first call.